Support typescript 6

TypeScript 6 is not allowed due to strict peer dependency

prisma-json-types-generator@4.1.1 requires:
  typescript@"^5.9.3"

This prevents installation in projects using TypeScript 6:

npm install fails with ERESOLVE error.

TypeScript 6 is already compatible with the ecosystem, so this peer dependency is too restrictive.

Please update peerDependencies to:

  "typescript": ">=5.9.3"

This would allow using the package with newer TypeScript versions without breaking installation.

--- a/.gitignore
+++ b/.gitignore
@@ -6,3 +6,4 @@ test/target/
 package-lock.json
 .DS_Store
 yarn.lock
+*.tsbuildinfo

--- a/README.md
+++ b/README.md
@@ -21,7 +21,7 @@
 <br />
 
 > [!IMPORTANT]  
-> **Prisma 7** is now supported with `prisma-json-types-generator@4` onwards! Prisma v6 is still supported in v4+.
+> `prisma-json-types-generator@5+` only supports Prisma v7 and TypeScript v6.
 
 <br />
 
@@ -300,7 +300,7 @@ It then traverses this AST, and for each property signature in a model, it cross
 ## Limitations
 
 - **Complex Filters:** To preserve functionality, types like `JsonFilter` and `JsonWithAggregatesFilter` remain untyped.
-- **Prisma Version:** This generator supports Prisma v5+ and generator v3+.
+- **Version Support:** `prisma-json-types-generator@5+` only supports Prisma v7 and TypeScript v6.
 - **Known Gaps:** If you find any `Json` fields that are not being typed correctly, please [open an issue](https://github.com/arthurfiorette/prisma-json-types-generator/issues).
 
 <br />

--- a/index.js
+++ b/index.js
@@ -6,11 +6,4 @@
  * https://github.com/arthurfiorette/prisma-json-types-generator
  */
 
-// Allows error stack traces to be readable if available
-try {
-  if (process.sourceMapsEnabled === false) {
-    process.setSourceMapsEnabled(true);
-  }
-} catch {}
-
 require('./dist/generator.js');

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "prisma-json-types-generator",
-  "version": "4.1.1",
+  "version": "5.0.0",
   "description": "Changes JsonValues to your custom typescript type",
   "keywords": [
     "prisma",
@@ -38,30 +38,35 @@
     "test-unit": "pnpm tsx --test test/unit/**/*.test.ts"
   },
   "dependencies": {
-    "@prisma/generator-helper": "^7.2.0",
-    "semver": "^7.7.3",
+    "@prisma/generator-helper": "^7.8.0",
+    "semver": "^7.7.4",
     "try": "^1.0.3",
     "tslib": "^2.8.1"
   },
   "devDependencies": {
     "@arthurfiorette/biomejs-config": "^2.0.1",
-    "@biomejs/biome": "^2.3.8",
-    "@prisma/dmmf": "^7.1.0",
-    "@prisma/generator": "^7.2.0",
-    "@types/node": "^24.10.4",
+    "@biomejs/biome": "^2.4.13",
+    "@prisma/dmmf": "^7.8.0",
+    "@prisma/generator": "^7.8.0",
+    "@types/node": "^24.12.2",
     "@types/semver": "^7.7.1",
     "husky": "^9.1.7",
-    "prisma": "^7.2.0",
+    "prisma": "^7.8.0",
     "tsd": "^0.33.0",
     "tsx": "^4.21.0"
   },
   "peerDependencies": {
-    "@prisma/client": "^7.2.0",
-    "prisma": "^7.2.0",
-    "typescript": "^5.9.3"
+    "@prisma/client": "^7.8.0",
+    "prisma": "^7.8.0",
+    "typescript": "^6.0.3"
   },
-  "packageManager": "pnpm@10.27.0",
+  "packageManager": "pnpm@10.33.2",
   "engines": {
     "node": ">=14.0"
+  },
+  "tsd": {
+    "compilerOptions": {
+      "rootDir": "./"
+    }
   }
 }

--- a/pnpm-workspace.yaml
+++ b/pnpm-workspace.yaml
@@ -2,3 +2,12 @@ ignoredBuiltDependencies:
   - '@prisma/engines'
   - esbuild
   - prisma
+
+# Fails if trusted package stops being trusted
+trustPolicy: no-downgrade
+# Ignores trust policies for packages older than a week
+trustPolicyIgnoreAfter: 10080
+# Minimum of 1 day before released packages can be installed
+minimumReleaseAge: 1440
+# no git/tarball dependencies, only from trusted registry
+blockExoticSubdeps: true

--- a/src/handler/module.ts
+++ b/src/handler/module.ts
@@ -46,7 +46,9 @@ export function handlePrismaModule(
     // This allows some types to be generated even if others may fail
     // which is good for incremental development/testing
     if (error instanceof PrismaJsonTypesGeneratorError) {
-      return PrismaJsonTypesGeneratorError.handler(error);
+      // TODO: Better handler? investigate how to handle errors in the best way.
+      console.error(error);
+      continue;
     }
 
     // Stops this generator is error thrown is not manually added by our code.

--- a/src/on-manifest.ts
+++ b/src/on-manifest.ts
@@ -9,7 +9,9 @@ export function onManifest(): GeneratorManifest {
   try {
     const pkg = require('../package.json');
     const prismaPeerVersion = pkg.peerDependencies?.prisma;
+    const typescriptPeerVersion = pkg.peerDependencies?.typescript;
     const prismaVersion = require('prisma/package.json').version;
+    const typescriptVersion = require('typescript/package.json').version;
     version = pkg.version;
 
     if (prismaVersion && prismaPeerVersion && !semverSatisfies(prismaVersion, prismaPeerVersion)) {
@@ -17,6 +19,20 @@ export function onManifest(): GeneratorManifest {
         styleText(
           'red',
           `\n\nPrisma Json Types Generator@${version} relies on Prisma ${prismaPeerVersion} and was not tested with your Prisma@${prismaVersion} installation.
+The generated output might not work correctly or even completely break Prisma Client types.\n`
+        )
+      );
+    }
+
+    if (
+      typescriptVersion &&
+      typescriptPeerVersion &&
+      !semverSatisfies(typescriptVersion, typescriptPeerVersion)
+    ) {
+      console.log(
+        styleText(
+          'red',
+          `\n\nPrisma Json Types Generator@${version} relies on TypeScript ${typescriptPeerVersion} and was not tested with your TypeScript@${typescriptVersion} installation.
 The generated output might not work correctly or even completely break Prisma Client types.\n`
         )
       );

--- a/src/util/create-signature.ts
+++ b/src/util/create-signature.ts
@@ -14,9 +14,8 @@ export function createType(
   //   '@DtoCastType(Customization, ../../types, Customization)\n' +
   //   '[CustomizationType]'
   // So we need to find the correct line to parse
-  const lines = description.split('\n');
   let parsed = null;
-  for (const line of lines) {
+  for (const line of description.split('\n')) {
     parsed = parseTypeSyntax(line);
     if (parsed) {
       break;

--- a/src/util/error.ts
+++ b/src/util/error.ts
@@ -7,9 +7,4 @@ export class PrismaJsonTypesGeneratorError extends Error {
     super(message);
     Object.assign(this, data);
   }
-
-  // TODO: Better handler? investigate how to handle errors in the best way.
-  static handler(error: PrismaJsonTypesGeneratorError) {
-    console.error(error);
-  }
 }

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -25,14 +25,16 @@
     // "moduleDetection": "auto",                        /* Control what method is used to detect module-format JS files. */
 
     /* Modules */
-    "module": "CommonJS" /* Specify what module code is generated. */,
-    // "rootDir": "./",                                  /* Specify the root folder within your source files. */
-    "moduleResolution": "node" /* Specify how TypeScript looks up a file from a given module specifier. */,
+    "module": "nodenext" /* Specify what module code is generated. */,
+    "rootDir": "./src" /* Specify the root folder within your source files. */,
+    "moduleResolution": "nodenext" /* Specify how TypeScript looks up a file from a given module specifier. */,
     // "baseUrl": "./",                                  /* Specify the base directory to resolve non-relative module names. */
     // "paths": {},                                      /* Specify a set of entries that re-map imports to additional lookup locations. */
     // "rootDirs": [],                                   /* Allow multiple folders to be treated as one when resolving modules. */
     // "typeRoots": [],                                  /* Specify multiple folders that act like './node_modules/@types'. */
-    // "types": [],                                      /* Specify type package names to be included without being referenced in a source file. */
+    "types": [
+      "node"
+    ] /* Specify type package names to be included without being referenced in a source file. */,
     // "allowUmdGlobalAccess": true,                     /* Allow accessing UMD globals from modules. */
     // "moduleSuffixes": [],                             /* List of file name suffixes to search when resolving a module. */
     "resolveJsonModule": true /* Enable importing .json files. */,

